packages/mui-system/src/cssVars/useCurrentColorScheme.ts

```diff
@@ -173,8 +178,10 @@ export default function useCurrentColorScheme<SupportedColorScheme extends strin
         }
         const newMode = mode ?? defaultMode;
         try {
-          localStorage.setItem(modeStorageKey, newMode);
-        } catch {
+          if (modeStorage) {
+            modeStorage.set(newMode);
+          }
+        } catch (error) {
           // Unsupported
         }
         return {
```

Comment:
`modeStorage` already has a try/catch on its `set` method, so this should be equivalent:

```ts
if (modeStorage) {
    modeStorage.set(newModa)
}
```

Without the redundant try/catch, or am I missing something?

Reply:
Updated